src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEEventSymbol.cs

```diff
@@ -92,11 +92,16 @@ private enum Flags : byte
                 _eventType = metadataDecoder.GetTypeOfToken(eventType);
             }
 
+            TypeSymbol originalEventType = _eventType;
+            _eventType = DynamicTypeDecoder.TransformType(_eventType, 0, handle, moduleSymbol);
+            // Dynamify object type if necessary
+            _eventType = _eventType.AsDynamicIfNoPia(_containingType);
```

Comment:
That is correct, there could be custom modifiers inside the type. Amount of custom modifiers at the top level is passed through special parameter.